docs/proposals/remote-artifacts.md

```diff
@@ -0,0 +1,293 @@
+<!--
+Copyright The Shipwright Contributors
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+---
+title: remote-artifacts
+authors:
+  - @otaviof
+reviewers:
+  - @SaschaSchwarze0
+  - @qu1queee
+  - @adambkaplan
+approvers:
+  - TBD
+creation-date: 2020-10-02
+last-updated:  2020-10-23
+status: provisional
+---
+
+Shipwright Remote Artifacts
+---------------------------
+
+# Summary
+
+Remote artifacts are dependencies of the software building process, they represent binaries or
+other data stored outside of the version control system (Git). Hence, they are required when
+dealing with container images as well, and this enhancement proposal focuses on adding a remote
+artifacts support to Shipwright's Build API.
+
+# Motivation
+
+Give Shipwright's operator broader build use-case support by enhancing its capabilities to include
+the concept of Artifacts. In other words, remote entities that will be available for the image
+build process.
+
+End users will be able to include remote artifacts as a build runtime dependency, artifacts
+controlled by remote systems will be downloaded before the build process starts. Those artifacts may
+be a pre-compile binary, `jar` files, `war` files, etc.
+
+## Goals
+
+*   Provide means to declare remote artifacts, dependencies that can be employed on builds;
+*   Create the mechanism to download and prepare remote artifacts for builds;
+
+## Non-Goals
+
+*   Automate the upload of local artifacts into the cluster;
+*   Manage remote artifacts;
+*   Amend container images created by the given `BuildStrategy`;
+*   Walking a remote directory tree, similarly to a `git checkout`;
+
+# Proposal
+
+The enhancement proposal is centered around the idea of declaring external dependencies, here
+called "artifacts", and being able to use them during the container image building process.
+
+## Build Artifacts
+
+The remote artifacts will be directly expressed in the body of a `Build` resource, as the
+following example:
+
+```yml
+---
+apiVersion: build.dev/v1alpha1
+kind: Build
+metadata:
+  name: license-file
+spec:
+  sources:
```

Comment:
I am wondering if `sources` really makes sense here.  I think of source as a git repository or something to that affect.  Maybe this should be something more along the lines of `dependencies` like you would list in a maven file?

Reply:
My vote is for `inputs` which would be a symmetric naming together with the `output` image that we already have in the Build.

Reply:
@SaschaSchwarze0 are you okay with `sources` and the path ahead to unify them later on?

Reply:
> @SaschaSchwarze0 are you okay with `sources` and the path ahead to unify them later on?

I am okay with `sources`, yes. (But I doubt we would ever change this in the future. ;-) )

Reply:
I´m concerned more than one person is not fully buying the `spec.sources` , what are the implications for this EP to move into `spec.inputs` rather than `spec.sources`, it will also keep the API clean for now. And in another EP we will deal with multiple gits, where we can tackle that modifications to `spec.source`.

Reply:
I'm okay to use `.spec.inputs` for this purpose. Do we have objections on `.spec.inputs`?

Reply:
not from my side :)

Reply:
@sbose78, @adambkaplan, @gabemontero, @coreydaley would you object to use `.spect.inputs` instead?